plugins/pyembed/pyembed.cpp

```diff
@@ -1188,9 +1199,19 @@ class PythonRowStream : public CInterfaceOf<IRowStream>
         checkPythonError();
         resultAllocator.set(_resultAllocator);
     }
+    ~PythonRowStream()
+    {
+        if (resultIterator)
+        {
+            checkThreadContext();
+            GILBlock b(threadContext->threadState);
```

Comment:
JAT, since it's dependent on thread variables, GILBlock could just call checkThreadContext() itself and reference threadContext->threadState directly.